kratos/utilities/math_utils.h

```diff
@@ -1539,6 +1547,277 @@ class MathUtils
         return is_converged;
     }
      
+    /**
+     * This function gives the Jacobi SVD of a given mxn matrix (m>=n), returns U,S; where A=U*S*V
+     * U and V are unitary, and S is a diagonal matrix. 
+     * Where s_i >= 0, and s_i >= s_i+1 (which means that the biggest number is the first one and the smallest the last one)
+     * TODO: This version is quite innefficient, look for a real and mathematical implementation (not the algorithm found in Wikipedia!!)
+     * @param InputMatrix: The matrix where perform the SVD 
+     * @param UMatrix: The unitary U matrix
+     * @param SMatrix: The diagonal S matrix
+     * @param VMatrix: The unitary V matrix
+     * @param Tolerance: The tolerance considered
+     * @return iter: The number of iterations
+     */
+     
+    static inline std::size_t SingularValueDecomposition(
+        const MatrixType& InputMatrix,
+        MatrixType& UMatrix,
+        MatrixType& SMatrix,
+        MatrixType& VMatrix,
+        const TDataType Tolerance = std::numeric_limits<double>::epsilon()
+        )
+    {
+        const SizeType m = InputMatrix.size1();
```

Comment:
Why assign instead of =?

Reply:
I thought it was the same

Reply:
Most probably it is the same, however I would use the = to avoid doing the same thing in different ways (imagine that  in a future we want to change matrix library for example, then it is useful if the same operator is used everywhere)